web_services/foxx/brightid5/schemas.js

```diff
@@ -133,16 +124,15 @@ schemas = Object.assign({
     }).label('Link ContextId'),
     joi.object({
       name: joi.string().valid('Sponsor').required().description('operation name'),
-      hash: joi.string().description('sha256 hash of the operation message used for generating signature'),
       contextId: joi.string().description('the contextId for the user that is being sponsored by context'),
       id: joi.string().description('brightid of the user that is being sponsored by context. This field is not provided by context owners who sponsor the user as they do not have users brightids. BrightID nodes that are trusted by context owners and have the private key that is used to spend sponsorships assigned to the context, will replace `contextId` by this field before sending this operation to blockchain'),
       app: joi.string().required().description('the app name that user is being sponsored by'),
       sig: joi.string().required().description('deterministic json representation of operation object signed by the private key shared between context owners and trusted node operators which enable them to spend sponsorships assigned to the context'),
+      timestamp: joi.number().required().description('milliseconds since epoch when the operation created'),
```

Comment:
What's the reason for requiring a timestamp for sponsor operations?

Reply:
I found out that 1hive sent sponsorship requests for some contextIds but they were failed because they had no sponsorships. They were not able to sponsor those contextIds anymore, because the operation was sent and `failed` once, and the `hash` was in `operationHashes` (which is used to prevent replay attacks).
Having timestamp allow apps to send sponsor requests again if first one failed. Also I think having `Sponsor`operation follow the same pattern with all other operations and having `timestamp` on all operations is better I think.
I don't remember why initially we didn't add timestamp on `Sponsor` operation.